66_HLSLBxDFTests/tests.h

```diff
@@ -0,0 +1,704 @@
+        maxCountFreq = 0.f;
+        maxIntFreq = 0.f;
+    }
+
+    double RLGamma(double a, double x) {
+        const double epsilon = 0.000000000000001;
+        const double big = 4503599627370496.0;
+        const double bigInv = 2.22044604925031308085e-16;
+        assert(a >= 0 && x >= 0);
+
+        if (x == 0)
+            return 0.0f;
+
+        double ax = (a * std::log(x)) - x - std::lgamma(a);
+        if (ax < -709.78271289338399)
+            return a < x ? 1.0 : 0.0;
+
+        if (x <= 1 || x <= a)
+        {
+            double r2 = a;
+            double c2 = 1;
+            double ans2 = 1;
+
+            do {
+                r2 = r2 + 1;
+                c2 = c2 * x / r2;
+                ans2 += c2;
+            } while ((c2 / ans2) > epsilon);
+
+            return std::exp(ax) * ans2 / a;
+        }
+
+        int c = 0;
+        double y = 1 - a;
+        double z = x + y + 1;
+        double p3 = 1;
+        double q3 = x;
+        double p2 = x + 1;
+        double q2 = z * x;
+        double ans = p2 / q2;
+        double error;
+
+        do {
+            c++;
+            y += 1;
+            z += 2;
+            double yc = y * c;
+            double p = (p2 * z) - (p3 * yc);
+            double q = (q2 * z) - (q3 * yc);
+
+            if (q != 0)
+            {
+                double nextans = p / q;
+                error = std::abs((ans - nextans) / nextans);
+                ans = nextans;
+            }
+            else
+            {
+                error = 1;
+            }
+
+            p3 = p2;
+            p2 = p;
+            q3 = q2;
+            q2 = q;
+
+            if (std::abs(p) > big)
+            {
+                p3 *= bigInv;
+                p2 *= bigInv;
+                q3 *= bigInv;
+                q2 *= bigInv;
+            }
+        } while (error > epsilon);
+
+        return 1.0 - (std::exp(ax) * ans);
+    }
+
+    double chi2CDF(double x, int dof)
+    {
+        if (dof < 1 || x < 0)
+        {
+            return 0.0;
+        }
+        else if (dof == 2)
+        {
+            return 1.0 - std::exp(-0.5 * x);
+        }
+        else
+        {
+            return RLGamma(0.5 * dof, 0.5 * x);
+        }
+    }
+
+    float32_t3 mapColor(float v, float vmin, float vmax)
+    {
+        float32_t3 c = float32_t3(1, 1, 1);
+        float diff = vmax - vmin;
+        v = clamp<float>(v, vmin, vmax);
+
+        if (v < (vmin + 0.25f * diff))
+        {
+            c.r = 0;
+            c.g = 4.f * (v - vmin) / diff;
+        }
+        else if (v < (vmin + 0.5f * diff))
+        {
+            c.r = 0;
+            c.b = 1.f + 4.f * (vmin + 0.25f * diff - v) / diff;
+        }
+        else if (v < (vmin + 0.75f * diff))
+        {
+            c.r = 4.f * (v - vmin - 0.5f * diff) / diff;
+            c.b = 0;
+        }
+        else
+        {
+            c.g = 1.f + 4.f * (vmin + 0.75f * diff - v) / diff;
+            c.b = 0;
+        }
+
+        return c;
+    }
```

Comment:
I'd add this to `nbl::hlsl::visualization::false_color` and keep our Sci-viz gradient there

@AnastaZIuk same thing for your IESes

Some color mappings have particular names in scientific literature https://docs.paraview.org/en/latest/ReferenceManual/colorMapping.html

https://pmc.ncbi.nlm.nih.gov/articles/PMC4959790/

https://www.youtube.com/watch?v=tJ6aEIB61hY & https://www.fabiocrameri.ch/colourmaps/

Reply:
It corresponds to `jet` colormap in matplotlib
https://www.mathworks.com/help/matlab/ref/jet.html